src/common/actions/webhook.js

```diff
@@ -0,0 +1,53 @@
+import 'isomorphic-fetch';
+import { createSnap } from './repository-input';
+
+export const WEBHOOK_FAILURE = 'WEBHOOK_FAILURE';
+const REQUEST_OPTIONS = {
+  method: 'POST',
+  headers: {
+    'Content-Type': 'application/json'
+  },
+  credentials: 'same-origin'
+};
+
+export function createWebhook(repositoryString) {
+  return (dispatch) => {
+    let settings = REQUEST_OPTIONS;
+    settings.body = JSON.stringify({
+      account: repositoryString.split('/')[0],
+      repo: repositoryString.split('/')[1]
+    });
+
+    return fetch(`${getBaseUrl()}/api/github/webhook`, settings)
+      .then((response) => {
+        return response.text();
+      })
+      .then((body) => {
+        let result = JSON.parse(body);
+        if (result.status && result.payload) {
+          if (result.status == 'success' || result.payload.code == 'github-already-created') {
+            // Treat pre-existing builds like a new build
+            dispatch(createSnap(repositoryString));
```

Comment:
Maybe webhook creation should be the last step? After snap is successfully created?